### ISSUE TITLE
Pass Array[Enum#Value] to SQL directly?

### ISSUE THREAD

**Issue by fm3** [scalableminds/webknossos#6736]
After #6734 the only type still passed to sql as a literal is Array[Enum#Value]. Is there a way to pass this as a value, i.e. implement a working setParameter for this type?

**Comment by normanrz** [comment]
Have you tried something like?
```
case class EnumArrayValue(v: List[Enum#Value]) extends SqlValue with SqlEscaping {
  override def setParameter(pp: PositionedParameters): Unit = pp.setObject(v.map(_.toString).toArray, Types.ARRAY)

  override def debugInfo: String = "{" + v.mkString(",") + "}"
}```

**Comment by fm3** [comment]
Yes, that is what I had. Unfortunately it yields `ERROR: column "settings_allowedmodes" is of type webknossos.tasktype_modes[] but expression is of type character varying[]`

**Comment by normanrz** [comment]
Maybe the placeholder needs an additional cast: `?::webknossos.tasktype_modes[]`?

**Comment by fm3** [comment]
Yes, that did the trick, good idea! :tada: I’ll include it in #6734
